wcomponents-theme/src/main/js/wc/file/util.js

```diff
@@ -0,0 +1,135 @@
+define(["wc/dom/uid", "wc/file/getMimeType", "wc/config"], function (uid, getMimeType, wcconfig) {
+	"use strict";
+
+	var instance = new FileUtil(),
+
+		/**
+		 * Used when checking the newly created file is named with the correct extension.
+		 * If it does not already match any in the array then the extension at index zero will be appended.
+		 **/
+		mimeToExt = {
+			"image/jpeg": ["jpeg", "jpg"],
+			"image/bmp": ["bmp"],
+			"image/png": ["png"],
+			"image/gif": ["gif"],
+			"image/x-icon": ["ico"],
+			"image/webp": ["webp"],
+			"image/svg+xml": ["svg"],
+			"image/tiff": ["tif", "tiff"],
+			"text/plain": ["bas", "c", "h", "txt", "log", "bat"],
+			"text/tab-separated-values": ["tsv"],
+			"text/richtext": ["rtx"],
+			"text/html": ["htm", "html", "stm"],
+			"text/xml": ["xml"],
+			"application/msword": ["doc", "dot"],
+			"application/vnd.ms-powerpoint": ["pot,", "pps", "ppt"],
+			"application/rtf": ["rtf"],
+			"application/pdf": ["pdf"],
+			"application/vnd.ms-excel": ["xla", "xlc", "xlm", "xls", "xlt", "xlw"]
+		};
+
```

Comment:
At the very least this should have JSDoc `@constructor`

Reply:
👍 added little jsdoc too